### ISSUE TITLE
Off by 1 error

### ISSUE THREAD

**Issue by Decos14** [Mavengence/GapStatistics#1]
The predict method will return the optimal # of clusters - 1. Then the plot function will plot with that incorrect number of clusters. A good example of this is:

from gapstatistics import GapStatistics
from sklearn.datasets import make_blobs

centers = [[0,0], [10,10]]
X = make_blobs(n_samples=200, centers=centers, n_features=2, cluster_std=1)
n_iterations = 30

gs = GapStatistics(distance_metric='minkowski', return_params=True)

optimum, params = gs.fit_predict(K=10, X=X[0])

gs.plot()

**Comment by Mavengence** [comment]
Hey, thanks for pointing that out. I updated to version 0.1.5 by fixing / updating this:

	1.	Distance Metric Calculation: Ensured correct distance metric handling
	2.	PCA Sampling: Corrected PCA transformations and sampling logic
	3.	Wk Calculation: Verified the calculation of Wk using appropriate scaling
	4.	Gap Statistics Calculation: Ensured correct computation of gap statistics
	5.	Plotting Adjustments: Removed cmap from plt.scatter and set n_init explicitly in KMeans
